src/wallet-redux.js

```diff
@@ -51,12 +51,18 @@ export function reducer(state, action) {
                 ];
             });
 
-            return Object.assign({}, state, {
+            const newState = Object.assign({}, state, {
                 hasContent: true,
                 // converts array to map with id as key
                 entries: new Map(newEntries)
             });
 
+            if (state.activeWalletId !== LEGACY && newEntries.length > 0 && !newState.entries.get(state.activeWalletId)) {
+                newState.activeWalletId = [ ...newState.entries.values() ][0].id;
+            }
+
+            return newState;
+
```

Comment:
Hmm. Those were already added in improved form by my PR. Please merge master.

Reply:
I don't much like merging master into feature branches.
Fixed it when merging feature branch into master.